### DISCUSSION TITLE
GPIO Interrupt

### DISCUSSION THREAD

**Discussion by Giorgio122** [stnolting/neorv32#1092] [Q&A] [answered]
Hello,
I am currently working on a C application that has to be executed by the neorv32. Looking at the documentation i did not find anything but just as a curiosity, how hard could it be to enable GPIO interrupts? Or maybe I misread the documentation and the GPIO can already be used in interrupt mode?
Thanks!

**Comment by DS-567** [comment, marked as answer]
I think you have to use the external interrupt controller (XIRQ) from the datasheet for interrupts and not the GPIO module.

**Reply by Giorgio122** [reply]
Thanks, completely missed that and I don't even know why...

**Reply by DS-567** [reply]
You're welcome. I probably made the same mistake 😂.
